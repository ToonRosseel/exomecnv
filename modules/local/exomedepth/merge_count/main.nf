// MERGE COUNT FILES
process COUNT_MERGE {
    tag "$meta.id $meta.chr"

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/coreutils:9.3' :
        'biocontainers/coreutils:9.3' }"
    conda "${moduleDir}/environment.yml"

    publishDir "$params.outdir/exomedepth/counts", mode: 'copy'

    input:
    tuple val(meta), path(files)

    output:
    tuple val(meta), path("*.txt"), emit:merge
    path "versions.yml", emit:versions

    script:
    def prefix = task.ext.prefix ?: "${meta.id}_${meta.chr}"
    """
    for file in $files; do
        if [ -f ${prefix}.txt ]; then
            paste ${prefix}.txt <(awk '{print \$5}' \$file) > temp_auto.txt
            mv temp_auto.txt ${prefix}.txt
            else
            cp \$file ${prefix}.txt
        fi
    done

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        paste: \$(paste --version | sed '1!d; s/paste (GNU coreutils) //')
        mv: \$(mv --version | sed '1!d; s/mv (GNU coreutils) //')
        cp: \$(cp --version | sed '1!d; s/cp (GNU coreutils) //')
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}_${meta.chr}"
    """
    touch ${prefix}.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        paste: \$(paste --version | sed '1!d; s/paste (GNU coreutils) //')
        mv: \$(mv --version | sed '1!d; s/mv (GNU coreutils) //')
        cp: \$(cp --version | sed '1!d; s/cp (GNU coreutils) //')
    END_VERSIONS
    """
}
