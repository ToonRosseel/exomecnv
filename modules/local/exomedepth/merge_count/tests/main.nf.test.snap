{
    "homo_sapiens - counts": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "Sample1",
                            "chr": "test",
                            "sam": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "fam": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "Sample1_test.txt:md5,911202604c02c755cdd6ac4b0bccfd79"
                    ]
                ],
                "1": [
                    "versions.yml:md5,a940eaf1baea599bcb19ef1d439225e6"
                ],
                "merge": [
                    [
                        {
                            "id": "Sample1",
                            "chr": "test",
                            "sam": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "fam": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "Sample1_test.txt:md5,911202604c02c755cdd6ac4b0bccfd79"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a940eaf1baea599bcb19ef1d439225e6"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-20T13:13:30.157585496"
    },
    "homo_sapiens - counts - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "Sample1",
                            "chr": "test",
                            "sam": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "fam": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "Sample1_test.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,a940eaf1baea599bcb19ef1d439225e6"
                ],
                "merge": [
                    [
                        {
                            "id": "Sample1",
                            "chr": "test",
                            "sam": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "fam": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "Sample1_test.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a940eaf1baea599bcb19ef1d439225e6"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-20T13:13:38.458248261"
    }
}